update Istio upgrade doc. (#3250)
istio/istio.io: content/docs/setup/kubernetes/upgrading-istio/index.md

1. [Download the new Istio release](/docs/setup/kubernetes/download-release/)
and change directory to the new release directory.

### Control plane upgrade

The Istio control plane components include: Citadel, Ingress gateway, Egress gateway, Pilot, Galley, Policy, Telemetry and
Sidecar injector.

#### Helm upgrade

If you installed Istio with [Helm](/docs/setup/kubernetes/helm-install/#option-2-install-with-helm-and-tiller-via-helm-install) the preferred upgrade option is to let Helm take care of the upgrade:

> 1. Upgrade the `istio-init` chart or install new release if it doesn't exists to keep all the Istio's [Custom Resource Definitions or CRDs for short](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions) up-to-date:

Reviewer: The text is difficult to understand because of an embedded conditional clause. I suggest splitting the sentence to make it easier to read. You find my suggestion below:
```suggestion
1. To keep updated all the Istio [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions) (CRDs), you must upgrade the `istio-init` chart. If the chart doesn't exist you must install the new release.
```
It would be better still if you could be specific or point to the release you mean the reader to install if the chart is not there.

Author: done

{{< text bash >}}
$ helm upgrade --install istio-init install/kubernetes/helm/istio-init --namespace istio-system
{{< /text >}}

> 1. Verify all the Istio's CRDs have been committed to the Kubernetes API server by checking all the CRD creation jobs complete with success:

Reviewer: This sentence is hard to understand, do you mean:
```suggestion
1. Check that all the CRD creation jobs completed successfully to verify that the Kubernetes API server received all the CRDs:
```

Author: done

> 1. [Download the new Istio release](/docs/setup/kubernetes/download-release/)

Reviewer: By removing the second step, you left this one orphaned. Just remove the number for it to render as normal text.
```suggestion
[Download the new Istio release](/docs/setup/kubernetes/download-release/)
```

You can also use Kubernetes’ rolling update mechanism to upgrade the control plane components. This is suitable for cases when Istio hasn't been installed using Helm.

> 1. Upgrade all the Istio's CRDs via `kubectl apply`, and wait a few seconds for the CRDs to be committed in the Kube api-server:

Reviewer: To make the actions the reader needs to take clearer, I suggest:
```suggestion
1. Use `kubectl apply` to upgrade all the Istio's CRDs.  Wait a few seconds for the Kubernetes API server to receive the upgraded CRDs:
```

Author: done

{{< text bash >}}
$ for i in install/kubernetes/helm/istio-init/files/crd*yaml; do kubectl apply -f $i; done
{{< /text >}}

> 1. Render Istio's core components to a Kubernetes manifest file, e.g.

Reviewer: It's unclear to the reader to identify what you mean with this step. I advise you change the verb used since "Render" might be too abstract. What exactly does the command do? Avoid using Latin abbreviations since they are difficult to understand too.
```suggestion
1. Add Istio's core components to a Kubernetes manifest file, for example.
```

Author: done

{{< text bash >}}
$ helm template install/kubernetes/helm/istio --name istio \
  --namespace istio-system > $HOME/istio.yaml
{{< /text >}}

> If you want to enable [global mutual TLS](/docs/concepts/security/#mutual-tls-authentication), set `global.mtls.enabled` and `global.controlPlaneSecurityEnabled` to `true` for the last command:

Reviewer: Since this text is part of the previous step, please indent the text and the command with four spaces to have the ordered list continue.

Author: done

{{< text bash >}}
$ helm template install/kubernetes/helm/istio --name istio --namespace istio-system \
  --set global.mtls.enabled=true --set global.controlPlaneSecurityEnabled=true > $HOME/istio-auth.yaml
{{< /text >}}

> If using Kubernetes versions prior to 1.9, you should add `--set sidecarInjectorWebhook.enabled=false`.

Reviewer: This text also needs to be indented then.

Author: done

> 1. Upgrade the Istio control plane components via the manifest, e.g.

Reviewer: ```suggestion
1. Upgrade the Istio control plane components via the manifest, for example:
```

Author: done
